plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,97 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+// NLP-ENGINE requires LINUX to be set for linux compiles.
+// The NLP-ENGINE needs to be updated to __linux__.
+#ifdef __linux__
+#define LINUX 1
+#endif
+
+#include "nlp_engine.h"
+#include "nlp_eng.hpp"
+
+static NLP_ENGINE *nlpEngine = NULL;
+static NLP_ENGINE *activeNlpEngine = NULL;
+static CriticalSection csNLP; 
+
+NLPEng::NLPEng() {}
+
+NLPEng::~NLPEng() {
+    CriticalBlock block(csNLP);
+    if (nlpEngine != NULL) {
+        delete nlpEngine;
```

Comment:
my comment above still stands, but if this code stays as it is it also needs to set nlpEngine to nullptr.